api/preprints/serializers.py

```diff
@@ -143,5 +149,41 @@ class PreprintDetailSerializer(PreprintSerializer):
     id = IDField(source='_id', required=True)
     subjects = JSONAPIListField(required=False, source='preprint_subjects')
 
+
 class PreprintDetailRetrieveSerializer(PreprintDetailSerializer):
     subjects = JSONAPIListField(required=False, source='get_preprint_subjects')
+
+
+class PreprintPreprintProviderRelationshipSerializer(ser.Serializer):
+    id = ser.CharField(source='_id', required=False, allow_null=True)
+    type = TypeField(required=False, allow_null=True)
+
+    links = LinksField({'self': 'get_self_url',
+                        'html': 'get_related_url'})
+
+    def get_self_url(self, obj):
+        return obj['self'].absolute_api_v2_url + 'relationships/preprints/'
+
+    def get_related_url(self, obj):
+        return obj['self'].absolute_api_v2_url + '/preprints/'
+
+    class Meta:
+        type_ = 'preprint_providers'
+
+    def make_instance_obj(self, obj):
+        return {
+            'data': obj.preprint_provider,
+            'self': obj
+        }
+
+    def update(self, instance, validated_data):
+        node = instance['self']
+        auth = get_user_auth(self.context['request'])
+
+        if node.preprint_provider:
+            raise ValueError('Preprint provider is already assigned to this preprint')
+
+        preprint_provider = PreprintProvider.load(validated_data['_id'])
+        node.set_preprint_provider(preprint_provider, auth, save=True)
```

Comment:
Is there any reason not to include this in `PreprintSerializer`'s `.create` or `.update`?